polaris-react/src/utilities/use-event-listener.ts

```diff
@@ -0,0 +1,31 @@
+import {RefObject, useEffect, useRef} from 'react';
+
+type ElementType = Window | Document | HTMLElement | RefObject<HTMLElement>;
```

Comment:
```suggestion
type ElementType = Window | Document | HTMLElement;
```

Suggest removing as we can emulate the same behavior by passing in `ref.current` and also I was having type inferences issues using refs in my POC e.g.
```tsx
useEventListener('click', handler, myRef.current)
```